internal/generate/packagemanifest/packagemanifest_test.go

```diff
@@ -0,0 +1,272 @@
+	"sigs.k8s.io/yaml"
+
+	genutil "github.com/operator-framework/operator-sdk/internal/generate/internal"
+)
+
+var (
+	testDataDir = filepath.Join("..", "testdata")
+)
+
+func TestGenerator(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Generator Suite")
+}
+
+var pkgManDefault, pkgManSingleChannel *apimanifests.PackageManifest
+
+var _ = BeforeSuite(func() {
+	initTestPackageManifestsHelper()
+})
+
+var _ = Describe("Generating a PackageManifest", func() {
+	format.UseStringerRepresentation = true
+
+	var (
+		g            Generator
+		buf          *bytes.Buffer
+		operatorName = "memcached-operator"
+		version      = "0.0.1"
+	)
+
+	BeforeEach(func() {
+		buf = &bytes.Buffer{}
+	})
+
+	Describe("for the new Go project layout", func() {
+
+		Context("with correct Options", func() {
+
+			var (
+				tmp string
+				err error
+			)
+
+			BeforeEach(func() {
+				tmp, err = ioutil.TempDir(".", "")
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			AfterEach(func() {
+				if tmp != "" {
+					os.RemoveAll(tmp)
+				}
+			})
+
+			It("should write a PackageManifest to an io.Writer", func() {
+				g = Generator{
+					OperatorName: operatorName,
+					Version:      version,
+				}
+				opts := []Option{
+					WithBase(testDataDir),
+					WithWriter(buf),
+				}
+				Expect(g.Generate(opts...)).ToNot(HaveOccurred())
+				Expect(buf.String()).To(MatchYAML(pkgManDefaultContent))
+			})
+			It("should write a PackageManifest to a file", func() {
+				g = Generator{
+					OperatorName: operatorName,
+					Version:      version,
+				}
+				opts := []Option{
+					WithBase(testDataDir),
+					WithFileWriter(tmp),
+				}
+				Expect(g.Generate(opts...)).ToNot(HaveOccurred())
+				outputFile := filepath.Join(tmp, makePkgManFileName(operatorName))
+				Expect(outputFile).To(BeAnExistingFile())
+				Expect(string(readFileHelper(outputFile))).To(MatchYAML(pkgManDefaultContent))
+			})
+		})
+
+		Context("with incorrect configuration", func() {
+
+			BeforeEach(func() {
+				g = Generator{
+					OperatorName: operatorName,
+					Version:      version,
+				}
+			})
+
+			It("should return an error without any Options", func() {
+				opts := []Option{}
+				Expect(g.Generate(opts...)).To(MatchError(errNoGetWriter))
+			})
+			It("should return an error without a getWriter", func() {
+				opts := []Option{
+					WithBase(testDataDir),
+				}
+				Expect(g.Generate(opts...)).To(MatchError(errNoGetWriter))
+			})
+			It("should return an error without a getBase", func() {
+				opts := []Option{
+					WithWriter(&bytes.Buffer{}),
+				}
+				Expect(g.Generate(opts...)).To(MatchError(errNoGetBase))
+			})
+			It("should return an error without a Version", func() {
+				g.Version = ""
+				opts := []Option{
+					WithBase(testDataDir),
+					WithWriter(&bytes.Buffer{}),
+				}
+				Expect(g.Generate(opts...)).To(MatchError(errNoVersion))
+			})
+		})
+
+		Context("to create a new package manifest file", func() {
```

Comment:
Shouldn't this be inside the first context `with correction configuration`?

Reply:
These tests are directly testing `Generator.generate()`, while the above contexts are testing `Options` passed to the generator.

Reply:
👍